### DISCUSSION TITLE
Is it possible to run multiple tests with different configs and get total coverage?

### DISCUSSION THREAD

**Discussion by tomo0613** [vitest-dev/vitest#3243] [Q&A] [answered]
I have two different kind of tests (lets say: snapshot and integration).
Currently I have two different vitest configs for them one for '*.test.' the other for '*.spec.' files,
(snapshots need a lot of mocking, so those have an extra `setupFile`)

I'd like to run them, and get coverage
seems to me that with "--coverage.enabled" would be able to do something similar
currently if I run "test:coverage", the coverage of the latest will be the result

`
"test-i": "npx vitest run -c ./vitest-integration.config.ts",
"test-i:coverage": "yarn test-i --coverage.enabled --coverage.all --coverage.src='./src' --coverage.reporter='text-summary' --coverage.reporter='lcov' ",
"test-s": "npx vitest run -c ./vitest-snapshot.config.ts",
"test-s:coverage": "yarn test-s --coverage.enabled --coverage.all --coverage.src='./src' --coverage.reporter='text-summary' --coverage.reporter='lcov' ",
"test:coverage": "yarn test-i:coverage && yarn test-s:coverage",
`

**Comment by sheremet-va** [comment, marked as answer]
Yeah, the last one wins and will overwrite the coverage. So you should either define different coverage folders for each project and merge coverages yourself. Or use [workspace feature](https://vitest.dev/guide/workspace.html) and run Vitest as a single command.

**Reply by tomo0613** [reply]
using workspace solved it.
Thank You.
